src/CashuWallet.ts

```diff
@@ -548,6 +548,8 @@ class CashuWallet {
 			} else {
 				secret = randomBytes(32);
 			}
+			secret = bytesToHex(secret);
+			secret = new TextEncoder().encode(secret);
```

Comment:
This might be a breaking change for NodeJS implementations, as TextEncoder is not part of the global scope iirc.

Reply:
PS: Why are we encoding bytes to hex and then reading hex string as utf8 to bytes again?

Reply:
I would suggest to use different variables with fixed types like `secret_bytes` and `secret_str`.

Reply:
> This might be a breaking change for NodeJS implementations, as TextEncoder is not part of the global scope iirc.
We're already using `TextEncoder`
https://github.com/cashubtc/cashu-ts/blob/bc6724aa0fae2454a338d22b85dd04c21b9ba6d3/src/utils.ts#L136

I wonder if NodeJS breaks, I haven't tested in a while. Also, isn't jest runing in the nodeJS context? It should break there also, no?
@Egge21M

Reply:
> What am I missing here?
1. we convert the bytes to a hex string
2. we encode the hex string to bytes

It's not the same bytes that are coming out as those that wen't in. It's a bit confusing, but thats mainly due to it being JS

Reply:
> 1. we convert the bytes to a hex string
> 
> 2. we encode the hex string to bytes
> 


> It's not the same bytes that are coming out as those that wen't in. It's a bit confusing, but thats mainly due to it being JS

It's not about JS. You encode an Arraybuffer into hex (e.g. 32 bytes = 64 hex characters) and then read that as utf-8 (e.g. 64 characters = 64 bytes)

Reply:
yes, this is according to spec: https://github.com/cashubtc/nuts/blob/main/00.md#proof